### PR TITLE
Add a loading indicator to Profile when loading data

### LINKED ISSUE #1632: Profile: add a loader when user details are updated
Currently, when we change accounts, profile data is updated. Sometimes, it might take a few seconds. 
https://watch.screencastify.com/v/8CFoOruYDtjdHMLIjEul

It would be nice to add a loader and ( gray out the page area?) when user's data is loading

### PR DESCRIPTION
**Pull request by ramirotw**
# Summary

Closes #1632 

This adds a loading indicator to Profile page labels using the shimmer effect. It also sets an interval to fetch profile data each 5 minutes.

  # To Test

1. Connect wallet and open /profile page
2. Labels should glow while profile is loading
3. Wait 5 minutes to validate profile data is fetched again

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @ramirotw , cool changes. 
However, loading effect is so quick, that it seems like blinking again. 
Could we add at least 2 sec of shimmering effect to make it noticeable (something like this https://codepen.io/andru255/pen/wvBdxbb , but for elements only) ?
See the video: when data is reloading, it shows dashes instead of numbers/date (23-26 sec of the video).

**Comment by ramirotw** [comment]
> See the video: when data is reloading, it shows dashes instead of numbers/date (23-26 sec of the video).

@elena-zh is it possible that the video link you referred to wasn't pasted?

**Comment by elena-zh** [comment]
@ramirotw , yes, it appeared to be so. 
Here it is https://watch.screencastify.com/v/s3YkMOFKbn1HokZA89YN

Sorry about that :(

File: src/custom/hooks/useFetchProfile.ts [hunk 1]
```diff
@@ -3,22 +3,40 @@ import { useEffect, useState } from 'react'
...
-export default function useFetchProfile() {
+type FetchProfileState = {
+  profileData: ProfileData | null
+  //error: string
+  isLoading: boolean
+}
+
+const emptyState: FetchProfileState = {
+  profileData: null,
+  //error: '',
```
Comment (alfetopito) [inline src/custom/hooks/useFetchProfile.ts:14]: Dead code?

File: src/custom/hooks/useFetchProfile.ts [hunk 2]
```diff
@@ -3,22 +3,40 @@ import { useEffect, useState } from 'react'
...
-  const [profileData, setProfileData] = useState<ProfileData | null>(null)
+  const [profile, setProfile] = useState<FetchProfileState>(emptyState)
 
   useEffect(() => {
     async function fetchAndSetProfileData() {
       if (chainId && account) {
+        setProfile({ ...emptyState, isLoading: true })
         const profileData = await getProfileData(chainId, account)
-        setProfileData(profileData)
+        setProfile({ ...emptyState, isLoading: false, profileData })
```
Comment (alfetopito) [inline src/custom/hooks/useFetchProfile.ts:29]: Any reason to reset the state on every load?

I would prefer to have the old value visible while it's loading

**Comment by anxolin** [comment]
Since we are deploying today, and this one has unaddressed issues, I will leave this one out.

We will be able to repoint this PR to develop branch later, please don't merge to the release/1.4.0 after we close the version

File: src/custom/hooks/useFetchProfile.ts (hunk 1 above)
Comment (ramirotw) [reply src/custom/hooks/useFetchProfile.ts:14]: I left that commented to hint Henry to put there the error in https://github.com/gnosis/cowswap/pull/1745

File: src/custom/hooks/useFetchProfile.ts (hunk 2 above)
Comment (ramirotw) [reply src/custom/hooks/useFetchProfile.ts]: Yes, good point, it even added an annoying flash between the dash and the value while loading

**Comment by ramirotw** [comment]
> See the video: when data is reloading, it shows dashes instead of numbers/date

Please re-test, this issue should be fixed. Later on if I find a suitable solution to delay the shimmer effect at the end I'll include it.

**Comment by ramirotw** [comment]
The build is failing due to an unrelated change:

> src/custom/pages/games/CowRunner/index.tsx
>  Line 4:10:  'useWalletInfo' is defined but never used  @typescript-eslint/no-unused-vars

**Comment by alfetopito** [comment]
> The build is failing due to an unrelated change:
> 
> > src/custom/pages/games/CowRunner/index.tsx
> > Line 4:10:  'useWalletInfo' is defined but never used  @typescript-eslint/no-unused-vars

Same thing that happened to @henrypalacios 's PR yesterday.

Couldn't find what was causing it but I assume it was a cache issue with the stored dependencies.

Will see if I can find out more about it.

**Comment by alfetopito** [comment]
> > The build is failing due to an unrelated change:
> > > src/custom/pages/games/CowRunner/index.tsx
> > > Line 4:10:  'useWalletInfo' is defined but never used  @typescript-eslint/no-unused-vars
> 
> Same thing that happened to @henrypalacios 's PR yesterday.
> 
> Couldn't find what was causing it but I assume it was a cache issue with the stored dependencies.
> 
> Will see if I can find out more about it.

In the case of this branch, it does indeed have the "broken" code https://github.com/gnosis/cowswap/blob/926a4406f4a057d23fa9c7fa6f462733ec519657/src/custom/pages/games/CowRunner/index.tsx#L4

And it is present on `develop` https://github.com/gnosis/cowswap/blob/develop/src/custom/pages/games/CowRunner/index.tsx

But it's not on `master` (https://github.com/gnosis/cowswap/blob/master/src/custom/pages/games/CowRunner/index.tsx) nor on `release/1.4.0` (https://github.com/gnosis/cowswap/blob/release/1.4.0/src/custom/pages/games/CowRunner/index.tsx)

Probably a bad merge.

I'll send a PR to `develop` to fix it.

**Comment by alfetopito** [comment]
@ramirotw this PR should have fixed it https://github.com/gnosis/cowswap/pull/1770
I already merged it to develop

**Comment by elena-zh** [comment]
Hey @ramirotw , I do not see a loading effect each 5 minutes when a profile data is updated... 
In addition, some new issues: 
1. There is no 'loading' effect when we change accounts (it was described in the main part of #1632 task)
2. There is no 'loading' effect when we connect a wallet, a data is loaded
3. When a profile's data loading fails, I see an eternal loading effect. Since we are going to show an error message for this case, I see no point in showing a loading effect for 5 minutes until the next page update/user's manual update. 
https://watch.screencastify.com/v/A06UhEQh0j81wAPiMNWh

**Comment by ramirotw** [comment]
> 1\.  There is no 'loading' effect when we change accounts (it was described in the main part of [Profile: add a loader when user details are updated #1632](https://github.com/gnosis/cowswap/issues/1632) task)
> 2\.  There is no 'loading' effect when we connect a wallet, a data is loaded

That's weird, I'm testing on https://pr1753--gpswapui.review.gnosisdev.com/#/profile and locally and I can see the shimmer effect, since the API responds quickly it doesn't take that long before the effect ends though.

> 3\. When a profile's data loading fails, I see an eternal loading effect. Since we are going to show an error message for this case, I see no point in showing a loading effect for 5 minutes until the next page update/user's manual update.
>     https://watch.screencastify.com/v/A06UhEQh0j81wAPiMNWh

I have added an error catch to turn off the loader

**Comment by elena-zh** [comment]
Thanks, @ramirotw , I have simulated a slow 3g connection and was able to see the loading effect on accounts changing/data updating. 
Loading effect after the error is also removed.

**Review by elena-zh — APPROVED** [review]

**Comment by elena-zh** [comment]
@ramirotw , I have created a separate task #1778 for you to make a 'loading' effect a bit longer

**Review by alfetopito — APPROVED** [review]

**Pull request merged**